src/NLog/Internal/Reflection/PropertyHelper.cs

```diff
@@ -103,21 +103,31 @@ internal static void SetPropertyFromString(object obj, string propertyName, stri
             try
             {
                 Type propertyType = propInfo.PropertyType;
+                var unwrappedType = UnwrapTypedLayout(propertyType);
+                var isTypedLayout = unwrappedType != propertyType;
 
-                if (!TryNLogSpecificConversion(propertyType, value, configurationItemFactory, out var newValue))
+                if (!TryNLogSpecificConversion(unwrappedType, value, configurationItemFactory, out var newValue))
                 {
                     if (propInfo.IsDefined(_arrayParameterAttribute.GetType(), false))
                     {
                         throw new NotSupportedException($"Parameter {propertyName} of {objType.Name} is an array and cannot be assigned a scalar value.");
                     }
 
-                    propertyType = Nullable.GetUnderlyingType(propertyType) ?? propertyType;
+                    unwrappedType = Nullable.GetUnderlyingType(unwrappedType) ?? unwrappedType;
 
-                    if (!(TryGetEnumValue(propertyType, value, out newValue, true)
-                        || TryImplicitConversion(propertyType, value, out newValue)
-                        || TryFlatListConversion(obj, propInfo, value, configurationItemFactory, out newValue)
-                        || TryTypeConverterConversion(propertyType, value, out newValue)))
-                        newValue = Convert.ChangeType(value, propertyType, CultureInfo.InvariantCulture);
+                    if (!(TryGetEnumValue(unwrappedType, value, out newValue, true)
+                          || TryImplicitConversion(unwrappedType, value, out newValue)
+                          || TryFlatListConversion(obj, propInfo, value, configurationItemFactory, out newValue)
+                          || TryTypeConverterConversion(unwrappedType, value, out newValue)))
+                    {
+                        newValue = Convert.ChangeType(value, unwrappedType, CultureInfo.InvariantCulture);
+                    }
+                }
+
+                // Wrap value again
+                if (isTypedLayout)
+                {
+                    newValue = WrapIntoTypedLayout(newValue, unwrappedType);
```

Comment:
What happens when `unwrappedType` has changed from nullable to non-nullable ? Will the Layout always render non-nullable?

Reply:
The problem where are fixing here (this should be in code as comment)

When using reflection:

- default conversations in .net are problematic to nullables (so convert to int instead of int?)
- set a int value to a nullable int property is also problematic. So therefore the wrap again. 

Does this answer your questions?

Reply:
No idea. I was just thinking that Nullable is a valid-return-value, but default-value of nullable-type is not the same.